_rules/table-headers-attribute-refer-to-data-cells-a25f45.md

```diff
@@ -0,0 +1,287 @@
+---
+id: a25f45
+name: Headers attribute of td element refers to cells in the same table element
+rule_type: atomic
+description: |
+  This rule checks that the headers attribute of `td` elements refer to cells in the same `table` element that have a semantic role of `columnheader` or `rowheader`.
+accessibility_requirements:
+  wcag20:1.3.1: # Info and Relationships (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+authors:
+  - Jey Nandakumar
+  - Audrey Maniez
+---
+
+## Applicability
+
+This rule applies to any [`headers`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/td#attr-headers) attribute specified on a [`cell`](https://www.w3.org/TR/html52/tabular-data.html#cell) in a [table](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table) element that is [included in the accessibility tree](#included-in-the-accessibility-tree).
+
+## Expectation
+
+The `headers` attribute of each target element is [a set of space separated IDs](https://www.w3.org/TR/html50/infrastructure.html#set-of-space-separated-tokens), each of which is an ID of an element in the same [document tree](https://www.w3.org/TR/dom41/#document-trees) or [shadow tree](https://www.w3.org/TR/dom41/#shadow-trees) as the applicable elment, that:
```

Comment:
You don't need to look at trees here. A cell in the same table (the second bullet) can't be in a different tree. This part should simply be:

> The `headers` attribute of each target element is [a set of space separated IDs](https://www.w3.org/TR/html50/infrastructure.html#set-of-space-separated-tokens), each of which is an ID of an element that:

Reply:
Allow me to abuse the DOM yet again:

```html
<table>
  <th>
    <div></div>
  </th>
  <td headers="foo"></td>
</table>

<script>
  const th = document.createElement("th");
  th.id = "foo";

  document
    .querySelector("div")
    .attachShadow({ mode: "open" })
    .appendChild(th);
</script>
```

Reply:
@WilcoFiers seeing this comment - https://github.com/act-rules/act-rules.github.io/pull/412#discussion_r318053483, it think it is best to emphasise document/ shadow tree. Wdyt?

Reply:
@kasperisager that `th` element isn't in the same tree as the td (see the definitions of tree)

Reply:
@WilcoFiers Bah, I copy pasted the wrong code:

```html
<table>
  <th>
    <div></div>
  </th>
  <td headers="foo"></td>
</table>

<script>
  const th = document.createElement("th");
  th.id = "foo";

  document
    .querySelector("div")
    .appendChild(th);
</script>
```

Reply:
Leaving usages of `document tree` and `shadow tree`.